packages/blob/src/client.ts

```diff
@@ -104,18 +151,93 @@ export const upload = createPutMethod<UploadOptions>({
       options.cacheControlMaxAge !== undefined
     ) {
       throw new BlobError(
-        'addRandomSuffix and cacheControlMaxAge are not supported in client uploads. Configure these options at the server side when generating client tokens.',
+        `${methodName} doesn't allow addRandomSuffix and cacheControlMaxAge. Configure these options at the server side when generating client tokens.`,
       );
     }
-  },
-  async getToken(pathname: string, options: UploadOptions) {
-    const clientToken = await retrieveClientToken({
+  };
+}
+
+// client.multipartUpload
+
+export type ClientCreateMultipartUploadCommandOptions =
+  ClientCommonCreateBlobOptions & CommonUploadOptions;
+
+export const createMultipartUpload =
+  createCreateMultipartPutMethod<ClientCreateMultipartUploadCommandOptions>({
+    allowedOptions: ['contentType'],
+    extraChecks: createUploadExtraChecks('client/`createMultipartUpload`'),
+    async getToken(pathname, options) {
+      return retrieveClientToken({
+        handleUploadUrl: options.handleUploadUrl,
+        pathname,
+        clientPayload: options.clientPayload ?? null,
+        multipart: true,
+      });
+    },
+  });
+
+type ClientMultipartUploadCommandOptions = ClientCommonCreateBlobOptions &
+  CommonMultipartPutOptions &
+  CommonUploadOptions;
+
+export const multipartUpload =
+  createMultipartPutMethod<ClientMultipartUploadCommandOptions>({
+    allowedOptions: ['contentType'],
+    extraChecks: createUploadExtraChecks('client/`multipartUpload`'),
+    async getToken(pathname, options) {
+      return retrieveClientToken({
```

Comment:
this will fetch a new client token for each part. I'm not sure if we should do it like this or if we should only request a client token once in `createMultipartUpload` and reuse it for `upload` and `complete`